### PR TITLE
Optimize placement group log

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
Add to create placement group log, and optimize part of the code implementation.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc [hunk 1]
```diff
@@ -109,27 +110,25 @@ void GcsPlacementGroupManager::OnPlacementGroupCreationFailed(
 
 void GcsPlacementGroupManager::OnPlacementGroupCreationSuccess(
     std::shared_ptr<GcsPlacementGroup> placement_group) {
-  auto placement_group_id = placement_group->GetPlacementGroupID();
-  RAY_CHECK(registered_placement_groups_.count(placement_group_id) > 0);
+  RAY_LOG(WARNING) << "Successfully created placement group "
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_placement_group_manager.cc:113]: This should be INFO?

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc [hunk 2]
```diff
@@ -62,36 +62,35 @@ GcsPlacementGroupManager::GcsPlacementGroupManager(
...
+  // `CreatePlacementGroup` requests,
+  //  which will lead to resource leakage, we will solve it in next pr.
   // Mark the callback as pending and invoke it after the placement_group has been
   // successfully created.
-  placement_group_to_register_callbacks_[placement_group_id].emplace_back(
-      std::move(callback));
-  RAY_CHECK(registered_placement_groups_
-                .emplace(placement_group->GetPlacementGroupID(), placement_group)
-                .second);
+  placement_group_to_register_callbacks_[placement_group_id] = std::move(callback);
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_placement_group_manager.cc:81]: In this case, can we rename it to `placement_group_to_register_callback_ `?

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc (hunk 1 above)
Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_placement_group_manager.cc]: fixed

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc (hunk 2 above)
Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_placement_group_manager.cc]: fixed

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc [hunk 3]
```diff
@@ -154,29 +152,31 @@ void GcsPlacementGroupManager::HandleCreatePlacementGroup(
...
+          GCS_RPC_SEND_REPLY(send_reply_callback, reply, Status::OK());
+        });
+      }));
 }
 
 void GcsPlacementGroupManager::ScheduleTick() {
   reschedule_timer_.expires_from_now(boost::posix_time::milliseconds(500));
   reschedule_timer_.async_wait([this](const boost::system::error_code &error) {
-    if (error == boost::system::errc::operation_canceled) {
+    if (error == boost::asio::error::operation_aborted) {
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_placement_group_manager.cc:179]: Just out of curiosity. Are they different error code, or same?

**Review by rkooo567 — APPROVED** [review]

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc (hunk 3 above)
Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_placement_group_manager.cc:179]: The value of boost::system::errc::operation_canceled and boost::asio::error::operation_aborted is same. But It is recommended to use boost::asio::error::operation_aborted in the document of deadline_timer.

**Pull request merged**